flexboxAPIspecV2.md

```diff
@@ -0,0 +1,663 @@
+
+
+
+Flexbox Layout for WinUI 
+===
+
+# Background
+
+
+
+
+Flexbox Layout is designed to provide an efficient way to layout, align, and distribute space among items within a container, specifically when the size of the device is either unknown or dynamic.
+
+Flexbox is a widely used layout schematic in web development, it allows developers to quickly create complex layouts with a standardized system. By introducing Flexbox to WinUI we can increase the layout choices developers have and cater to those who desire to use this popular layout in their WinUI applications.
+
+More on Flexbox available here: https://css-tricks.com/snippets/css/a-guide-to-flexbox/#aa-flexbox-properties
+
+
+# Remarks
+
+
+
+WinUI Offers several Layout options, such as StackLayout and Flowlayout. Usage of Flexbox Layout will behave similarly and interact by wrapping your desired elements in a Layout Control.
+
+#### Flex Containers vs Flex Items
+
+Flex Containers are elements that hold and specify the arangement of Flex Items. Containers have the following accepted properties:
+| Property | 
+|-|
+| Direction |
+Wrap|
+JustifyContent|
+AlignItems|
+AlignContent|
+
+Flex Items are elements that are children of a Flex Container. Items have the following accepted properties:
+| Property | 
+|-|
+Order|
+Grow|
+Shrink|
+Basis|
+AlignSelf|
+Unit|
+
+
+
+#### Traditional CSS Behavior vs Flexbox in WinUI
+
+Several behaviors may deviate from the exact behavior found in CSS Flexbox, to highlight one in XAML, items do not overflow their containers, instead they will be omitted from the layout.
+
+
+
+# API Pages
+
+_(Each of the following L2 sections correspond to a page that will be on docs.microsoft.com)_
+
+
+
+## Flex Direction
+
+
+The flex-direction property specifies the direction of the flexible items.
+Note: If the element is not a flexible item, the flex-direction property has no effect.
+
+
+```xml
+<controls:LayoutPanel>
+    <controls:LayoutPanel.Layout>
+        <controls:FlexboxLayout Direction="Row"/>
+    </controls:LayoutPanel.Layout>
+            <Grid>
```

Comment:
To make this easier to read, leave out the Grids and just do the TextBlocks

Reply:
The grids in the examples are used to provide a way to apply styling. Since there is no "background color" property for Textblocks